### PR TITLE
UI fixes to dialog

### LINKED ISSUE #364: mySyncStashCheckbox visibility issues
mySyncStashCheckbox needs to made invisible when appropriate in CloudAttachDialog.checkSyncStashState

### PR DESCRIPTION
**Pull request by loosebazooka**
closes #364

### CODE REVIEW

**Comment by loosebazooka** [comment]
@elharo @joaoandremartins @patflynn whaddya thing?

File: google-cloud-tools-plugin/src/com/google/gct/idea/debugger/ui/CloudAttachDialog.java [hunk 1]
```diff
@@ -246,9 +251,6 @@ private void checkSyncStashState() {
       mySyncStashCheckbox.setText(
           GctBundle.getString("clouddebug.stash.local.changes.and.sync", mySyncResult.getTargetSyncSHA().substring(0, 7)));
       mySyncStashCheckbox.setSelected(true);
-      myWarningLabel.setVisible(false);
```
Comment (patflynn) [inline google-cloud-tools-plugin/src/com/google/gct/idea/debugger/ui/CloudAttachDialog.java]: these are not good label names aye. We should name them by the message they display.

File: google-cloud-tools-plugin/src/com/google/gct/idea/debugger/ui/CloudAttachDialog.java [hunk 2]
```diff
@@ -177,7 +176,7 @@ protected void doOKAction() {
       UsageTrackerProvider.getInstance()
           .trackEvent(GctTracking.CATEGORY, GctTracking.CLOUD_DEBUGGER, "start.session", null);
       // TODO : add source context tracking info
-      if (mySyncStashCheckbox.isSelected()) {
+      if (mySyncStashCheckbox.isVisible() && mySyncStashCheckbox.isSelected()) {
```
Comment (patflynn) [inline google-cloud-tools-plugin/src/com/google/gct/idea/debugger/ui/CloudAttachDialog.java]: if the checkbox is invisible, then we know that it's not selected don't we?

**Comment by elharo** [comment]
Can you describe these changes? Not immediately apparent to me from the code.

File: google-cloud-tools-plugin/src/com/google/gct/idea/debugger/ui/CloudAttachDialog.java [hunk 3]
```diff
@@ -111,7 +111,6 @@ public void actionPerformed(ActionEvent e) {
         if (mySyncStashCheckbox.isVisible()) {
           myWarningLabel.setVisible(!mySyncStashCheckbox.isSelected());
```
Comment (patflynn) [inline google-cloud-tools-plugin/src/com/google/gct/idea/debugger/ui/CloudAttachDialog.java]: how testable are these DialogWrapper subclasses?

A few tests to document what the warning and visibility and selected flags should be for each use-case could help people understand what this is up to.

File: google-cloud-tools-plugin/src/com/google/gct/idea/debugger/ui/CloudAttachDialog.java (hunk 2 above)
Comment (loosebazooka) [inline google-cloud-tools-plugin/src/com/google/gct/idea/debugger/ui/CloudAttachDialog.java]: oh this was from testing stuff and I didn't take it out, good catch.
I do think a checkbox can be invisible and checked though...

**Comment by loosebazooka** [comment]
The code is setting and unsetting visibility on the components in the dialog. I've fixed other bugs in here before. Anyway, the situation was that some code paths didn't reset visibility on UI elements correctly. So now it just sets them all to invisible and then enables them as necessary.

File: google-cloud-tools-plugin/src/com/google/gct/idea/debugger/ui/CloudAttachDialog.java (hunk 1 above)
Comment (loosebazooka) [inline google-cloud-tools-plugin/src/com/google/gct/idea/debugger/ui/CloudAttachDialog.java]: sure, probably in another CL though.

**Comment by elharo** [comment]
Do you know why the travis build hasn't completed?

File: google-cloud-tools-plugin/src/com/google/gct/idea/debugger/ui/CloudAttachDialog.java (hunk 1 above)
Comment (elharo) [inline google-cloud-tools-plugin/src/com/google/gct/idea/debugger/ui/CloudAttachDialog.java]: Good idea. Can you file a bug on this?

File: google-cloud-tools-plugin/src/com/google/gct/idea/debugger/ui/CloudAttachDialog.java [hunk 4]
```diff
@@ -301,10 +296,6 @@ else if (!mySyncResult.hasRemoteRepository()) {
       setOKButtonText(isContinued()
           ? GctBundle.getString("clouddebug.continuesession")
           : GctBundle.getString("clouddebug.attach"));
-      mySyncStashCheckbox.setVisible(false);
-      myWarningLabel.setVisible(false);
-      myWarningLabel2.setVisible(false);
-      myInfoPanel.setVisible(true);
```
Comment (elharo) [inline google-cloud-tools-plugin/src/com/google/gct/idea/debugger/ui/CloudAttachDialog.java]: myInfoPanel is now always visible, right?

**Comment by loosebazooka** [comment]
@etanshaul are you working on a fix that incorporates this?

**Comment by etanshaul** [comment]
I can incorporate it as part of #309. But I'm first fixing some smaller bugs with the dialog and setting up some test coverage around that class so that we can better define the expected behavior.  So if you wanted this in now, feel free to merge it.

**Comment by loosebazooka** [comment]
If it's not too much trouble, please go for it and I'll close this PR.

**Comment by loosebazooka** [comment]
closing, @etanshaul is taking care of this.

**Pull request closed without merging**